include/nix/DataArray.hpp

```diff
@@ -381,6 +382,84 @@ class NIXAPI DataArray : public base::EntityWithSources<base::IDataArray>, publi
         return dim;
     }
 
+    /**
+     * @brief Append a new {@link nix::DataFrameDimension} to the list of
+     *        dimension descriptors.  A DataFrameDimension can be used to
+     *        specify a dimension of a DataArray as specified by one (or all)
+     *        column(s) of the linked DataFrame. That is, for each entry in the
+     *        DataArray there must be a respecive row in the DataFrame. If no
+     *        column index is specified that whole DataFrame is assumed to
+     *        describe the DataArray entries along this dimension.
+     *
+     * @param frame: The DataFrame
+     * @param column_index: The index of the column that should be used.
+     *
+     * @return The new DataFrameDimension
+     */
+    DataFrameDimension appendDataFrameDimension(const DataFrame &frame, unsigned column_index) {
+        if (column_index > frame.columns().size()) {
+            throw nix::OutOfBounds("DataArray::appendDataFrameDimensios: Invalid columnIndex ", column_index);
+        }
+        if (!frame) {
+            throw nix::UninitializedEntity();
+        }
+
+        DataFrameDimension dim = backend()->createDataFrameDimension(backend()->dimensionCount() + 1, frame, column_index);
+        return dim;
+    };
+
+    /**
+     * @brief Append a new {@link nix::DataFrameDimension} to the list of
+     *        dimension descriptors.  A DataFrameDimension can be used to
+     *        specify a dimension of a DataArray as specified by one (or all)
+     *        column(s) of the linked DataFrame. That is, for each entry in the
+     *        DataArray there must be a respecive row in the DataFrame. If no
+     *        column index is specified that whole DataFrame is assumed to
+     *        describe the DataArray entries along this dimension.
+     *
+     * @param frame: The DataFrame
+     * @param column_name: The name of the column that should be used.
+     *
+     * @return The new DataFrameDimension
+     */
+    DataFrameDimension appendDataFrameDimension(const DataFrame &frame, const std::string &column_name) {
+        if (!frame) {
+            throw nix::UninitializedEntity();
+        }
+        unsigned index;
+        try {
+            std::vector<std::string> names = {column_name};
+            std::vector<unsigned> indices = frame.colIndex(names);
+            index = indices[0];
```

Comment:
There is `unsigned colIndex(const std::string &name) const` as a nonThere is `unsigned colIndex(const std::string &name) const` as a non-vector overload.